### PR TITLE
feat(detr): add RFDETR.evaluate() for COCO metrics on a dataset split

### LINKED ISSUE #1110: How to evaluate a trained model and print indicators？
Hello, I have trained the model now. How can I output the evaluation indicators of the test set? Is there a script？

### PR DESCRIPTION
**Pull request by anatoly-ryabchenko**
## What does this PR do?

Add a public `model.evaluate(split="test"|"val", **kwargs)` that runs a single evaluation pass via the PyTorch Lightning stack and returns (and prints) the COCO metrics computed by `COCOEvalCallback` (mAP, mAR, macro-F1, per-class AP). It evaluates the weights already held in memory — works after `train()` and after `from_checkpoint()` — and never passes `ckpt_path`, so it avoids the PTL loop-state-restore `KeyError` users may have otherwise hit when pointing `trainer.test(ckpt_path=...)` at a bare `.pth`.

Apart from `split`, `evaluate()` accepts exactly the same keyword arguments as `train()` (`dataset_dir`, `device`, `resolution`, `batch_size`, ...). To guarantee that parity without duplication, the shared kwarg preprocessing (device mapping, resolution override, deprecated-knob absorption, auto-batch, model_name) is extracted into `_prepare_run_config`, used by both `train()` and `evaluate()`. `evaluate()` does not adapt the detection head to the dataset; a class-count mismatch warns and proceeds unchanged. A `resolution` override is reconciled against the loaded weights via PE interpolation.

Supporting changes:
- `build_trainer(..., include_training_callbacks=False)` builds a lean eval-only trainer: keeps the metric callback (and progress bar) but no EMA, drop-path, checkpointing, best-model, early stopping, or loggers, and disables checkpointing, so evaluation writes nothing to `output_dir`.
- `COCOEvalCallback` now resolves per-class names in `on_test_start` / `on_validation_start` (not only the fit-only `on_fit_start`), so a standalone test/validate run labels per-class AP by class name.

**Related Issue(s):** Closes #1110

## Type of Change

- New feature (non-breaking change that adds functionality)

## Testing

- [x] I have tested this change locally
- [x] I have added/updated tests for this change

**Test details:**
#### `TestEvaluateReturnValue` (real end-to-end run, shared module fixture)
- `test_returns_dict` — return value is a dict
- `test_contains_map_metric` — `test/mAP_50_95` key present
- `test_contains_f1_metric` — `test/F1` key present
- `test_metric_values_are_finite_floats` — every metric is a finite scalar

#### `TestEvaluateDoesNotMutateModel`
- `test_num_classes_unchanged` — `model_config.num_classes` identical before/after
- `test_weights_unchanged` — module weights byte-for-byte identical after eval

#### `TestEvaluateClassNames`
- `test_per_class_keys_use_names` — per-class AP keys labelled by name (regression for the `COCOEvalCallback` test-only fix)

#### `TestEvaluateNoSideEffects`
- `test_no_checkpoint_files` — no `*.ckpt` written
- `test_no_metrics_csv` — no `metrics.csv` written

#### `TestEvaluateSplitDispatch` (mocked PTL stack)
- `test_split_test_calls_trainer_test` — `split="test"` → `trainer.test`
- `test_split_val_calls_trainer_validate` — `split="val"` → `trainer.validate`
- `test_invalid_split_raises` — invalid split → `ValueError`

#### `TestEvaluateKwargParity`
- `test_shares_train_kwarg_handling` — deprecated `do_benchmark` absorbed + warned (proves shared `_prepare_run_config`)

#### `TestEvaluateClassCountMismatch`
- `test_mismatch_warns` — dataset class count ≠ model → `UserWarning`, head unchanged

#### `TestEvaluateImportGuard`
- `test_missing_training_extra_raises_install_hint` — missing extra → `ImportError` with install hint
- `test_internal_training_module_import_error_preserved` — internal `rfdetr.*` failure → original `ModuleNotFoundError` preserved

#### `TestEvaluateTrainerBoundary`
- `test_device_index_forwarded_and_eval_mode` — `device="cuda:1"` → `build_trainer(accelerator="gpu", devices=[1], include_training_callbacks=False)`
- `test_empty_results_returns_empty_dict` — empty trainer results → `{}`

#### `TestEvaluateResolutionOverride`
- `test_resolution_override_evaluates` — `resolution=` override reconciles PE so the in-memory transplant loads and metrics return

#### `TestBuildTrainerEvalMode` class added, 6 tests
- `test_coco_eval_callback_present` — eval mode keeps `COCOEvalCallback`
- `test_no_checkpoint_callbacks` — no `ModelCheckpoint` (incl. `BestModelCallback`)
- `test_no_ema_callback` — no EMA callback even with `use_ema=True`
- `test_no_early_stopping_callback` — no early stopping even with `early_stopping=True`
- `test_loggers_disabled` — `trainer.loggers == []`
- `test_training_callbacks_present_by_default` — default path unchanged (`BestModelCallback` still wired)

#### `TestBuildTrainerLoggers` — 3 tests added to the existing class (Codecov patch-coverage follow-up)
- `test_wandb_logger_wired` — `WandbLogger` added when `wandb=True` (dep mocked)
- `test_missing_wandb_dep_warns_not_crashes` — missing wandb package → warning, CSV logger still present
- `test_missing_mlflow_dep_warns_not_crashes` — missing mlflow package → warning, CSV logger still present

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have updated the documentation accordingly (if applicable)

### CODE REVIEW

File: src/rfdetr/detr.py [hunk 1]
```diff
@@ -775,6 +811,104 @@ def train(self, **kwargs):
...
+        # already-loaded in-memory weights into it (the reverse of train()'s final
+        # `self.model.model = module.model`).  This evaluates the current weights and
+        # never passes ``ckpt_path``, sidestepping PTL's loop-state restore on a bare .pth.
+        eval_model_config = self.model_config.model_copy(update={"pretrain_weights": None})
+        module = RFDETRModelModule(eval_model_config, config)
+        source_state = self.model.model.state_dict()
+        # Reconcile DINOv2 positional embeddings when a `resolution` override changed the PE grid
+        # (no-op when unchanged), so the transplant works at the evaluation resolution.
+        interpolate_position_embeddings(source_state, eval_model_config.positional_encoding_size)
+        module.model.load_state_dict(source_state)
```
[the remark this answers is not in the document]
Comment (anatoly-ryabchenko) [reply src/rfdetr/detr.py]: I don't think that is true? If I create a model on cpu, take its state dict, convert model to gpu and load the cpu state dict (explicitly verifying it was on cpu), and do a forward pass, I don't have any errors. Vice versa also works...

**Review by Borda — DISMISSED** [review]
